### PR TITLE
Better mock of bot.event_manager.emit

### PR DESCRIPTION
**Pull request by th3w4y**
https://github.com/PokemonGoF/PokemonGo-Bot/issues/5339

### CODE REVIEW

**Comment by cmezh** [comment]
Now test runs in 0.007s. But maybe it's better to fix not the test, but FakeBot?

**Comment by th3w4y** [comment]
@cmezh  hehe i already started :)

**Comment by cmezh** [comment]
@th3w4y nice ;) then we shouldn't merge this PR for now.

**Comment by th3w4y** [comment]
Alldough there is not much usage of FakeBot() everywhere we use MagicMock() except:

```
pokemongo_bot/test/log_interval_test.py:from tests import FakeBot
pokemongo_bot/test/log_interval_test.py:        self.bot = FakeBot()
pokemongo_bot/test/sleep_schedule_test.py:from tests import FakeBot
tests/location_parser_test.py:from tests import FakeBot
tests/location_parser_test.py:        self.bot = FakeBot()
tests/update_live_stats_test.py:from tests import FakeBot
tests/update_live_stats_test.py:        self.bot = FakeBot()
```

File: pokemongo_bot/test/sleep_schedule_test.py [hunk 1]
```diff
@@ -22,12 +22,17 @@ class SleepScheduleTestCase(unittest.TestCase):
...
-        self.bot = FakeBot()
+        self.bot = MagicMock()
         self.worker1 = SleepSchedule(self.bot, self.config1)
         self.worker2 = SleepSchedule(self.bot, self.config2)
 
+        def emit(task, sender, formatted, data):
+            pass
+
+        self.bot.event_manager = MagicMock()
+        self.bot.event_manager.emit = emit
```
Comment (cmezh) [inline pokemongo_bot/test/sleep_schedule_test.py]: You can use ```self.bot.event_manager.emit = lambda: x``` and remove ```def emit``` (just a little code cleanup)

Comment (th3w4y) [inline pokemongo_bot/test/sleep_schedule_test.py]: lambda takes on argument we need 4

Comment (th3w4y) [inline pokemongo_bot/test/sleep_schedule_test.py]: `self.bot.event_manager.emit = lambda *args, **kwargs: None`

**Comment by cmezh** [comment]
:+1:

**Pull request merged**